.github/workflows/run-upstream-tests.yml

```diff
@@ -0,0 +1,71 @@
+name: run-upstream-tests
+
+description: |
+  Run upstream tests against the patched version of the upstream (terraform-provider-aws) provider.
+
+  This job will run acceptance tests that do provision actual AWS resources.
+
+  One notable problem is that tag-related tests do not currently pass for the patched provider, since the Pulumi bridged
+  provider handles tagging at a higher level. These tests are currently skipped.
+
+  The job matrix is organized by service, corresponding to `ls ./upstream/internal/service` folders.
+
+  Note that adding new services needs to be done with care because test failures may leave unsupervised resources in the
+  test account causing an increase in burn rate, so that account cleaners need to be extended for each new service.
+
+  The primary goal for additional coverage is testing services that have Pulumi patches that may accidentally regress
+  desired upstream behavior.
+
+on:
+  pull_request:
+    paths-ignore:
+    - CHANGELOG.md
+
+env:
+  PR_COMMIT_SHA: ${{ github.event.client_payload.pull_request.head.sha }}
+  AWS_REGION: us-west-2
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  TF_APPEND_USER_AGENT: pulumi
+
+# This should cancel any previous runs of the same workflow on the same branch which are still running.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```

Comment:
We probably need this, but it will also lead to more orphaned resources if tests are cancelled before the destroy step.

Reply:
Added a note 🙏